### ISSUE TITLE
[distributed]: Make bucket failure.

### ISSUE THREAD

**Issue by hackintoshrao** [minio/minio#3541]
Distributed Minio running 4 instances on one node fails to create bucket. But however its works with 4 instances minio run using `docker-compose`.
## Expected Behavior

Bucket creation to be successful.
## Current Behavior

- Hangs indefinitely.
## Possible Solution

- Check for deadlocks.
## Steps to Reproduce (for bugs)

1. Build from latest master.
2. Run 4 instances of minio on one node.
    -  ./minio server --address 127.0.0.1:9996 http://127.0.0.1:9996/tmp/rao1 http://127.0.0.1:9997/tmp/rao2 http://127.0.0.1:9998/tmp/rao3 http://127.0.0.1:9999/tmp/rao4
    - ./minio server --address 127.0.0.1:9997 http://127.0.0.1:9996/tmp/rao1 http://127.0.0.1:9997/tmp/rao2 http://127.0.0.1:9998/tmp3 http://127.0.0.1:9999/tmp/rao4
    - ./minio server --address 127.0.0.1:9998 http://127.0.0.1:9996/tmp/rao1 http://127.0.0.1:9997/tmp/rao2 http://127.0.0.1:9998/tmp/r3 http://127.0.0.1:9999/tmp/rao4
    - ./minio server --address 127.0.0.1:9999 http://127.0.0.1:9996/tmp/rao1 http://127.0.0.1:9997/tmp/rao2 http://127.0.0.1:9998/tmp/r3 http://127.0.0.1:9999/tmp/rao4

3. `mc mb mymino/test`

## Context

Testing Minio distributed.
## Your Environment

* Version used: Latest master at f37f56ac15e92e353dac135d8cbaacaba8202646 .

**Comment by krishnasrinivas** [comment]
you are using  http://127.0.0.1:9998/tmp/r3 instead of  http://127.0.0.1:9998/tmp/rao3 on 3rd and 4th server

**Comment by hackintoshrao** [comment]
Let me validate that. If it works then the bug would be to give better error message and not to start the server in that scenario.

**Comment by awwalker** [comment]
Also http://127.0.0.1:9998/tmp3 on the second...just in case that one was missed too

**Comment by harshavardhana** [comment]
> Let me validate that. If it works then the bug would be to give better error message and not to start the server in that scenario

Server command line validation has a separate bug #3475 - closing this bug.

**Comment by hackintoshrao** [comment]
The original issue exists, have the setup where `mc mb` is hung.

**Comment by harshavardhana** [comment]
> The original issue exists, have the setup where mc mb is hung.

It doesn't the configuration is wrong here is the script that works fine 

```
#!/bin/bash

$(minio server --address 65.19.167.92:9996 http://65.19.167.92:9996/tmp/1 http://65.19.167.92:9997/tmp/2 http://65.19.167.92:9998/tmp/3 http://65.19.167.92:9999/tmp/4)&
$(minio server --address 65.19.167.92:9997 http://65.19.167.92:9996/tmp/1 http://65.19.167.92:9997/tmp/2 http://65.19.167.92:9998/tmp/3 http://65.19.167.92:9999/tmp/4)&
$(minio server --address 65.19.167.92:9998 http://65.19.167.92:9996/tmp/1 http://65.19.167.92:9997/tmp/2 http://65.19.167.92:9998/tmp/3 http://65.19.167.92:9999/tmp/4)&
minio server --address 65.19.167.92:9999 http://65.19.167.92:9996/tmp/1 http://65.19.167.92:9997/tmp/2 http://65.19.167.92:9998/tmp/3 http://65.19.167.92:9999/tmp/4
``` 

This is with current master.

**Comment by harshavardhana** [comment]
You must be using an older binary or from shared backend change, which has a bug @hackintoshrao

**Comment by harshavardhana** [comment]
> You must be using an older binary or from shared backend change, which has a bug @hackintoshrao

Looks like the shared backend PR has the problem. Fixing it.

**Comment by hackintoshrao** [comment]
@harshavardhana : I'm able to produce with server built from latest master .

**Comment by harshavardhana** [comment]
> @harshavardhana : I'm able to produce with server built from latest master .

Fixed the shared backend PR as well. I don't see how you can reproduce with latest master.

**Comment by hackintoshrao** [comment]
Let me try again with latest master
